source/docs/essentials/create-a-ui-component.md

```diff
@@ -0,0 +1,307 @@
+>   [official Storybook documentation](https://storybook.js.org/basics/introduction/)
+> - about [our Storybook usage](#TODO) by reading our documentation
+
+### Step 3: Implement your component
+
+Now that everything is ready to go, you can do the actual work and implement the
+component. In the case of the `IllustratedContent`, it would look like this:
+
+```jsx
+// src/web/theme/ui/molecules/IllustratedContent/IllustratedContent.js
+import React from "react";
+import PropTypes from "prop-types";
+
+const IllustratedContent ({ media, children }) => (
+  <div className="illustrated-content">
+    <div className="illustrated-content__media">{media}</div>
+    <div className="illustrated-content__content">{children}</div>
+  </div>
+);
+
+IllustratedContent.propTypes = {
+  media: PropTypes.node.isRequired,
+  children: PropTypes.node.isRequired
+}
+
+export default IllustratedContent
+```
+
+#### Styling
+
+If you need to style things with CSS, you have to import the new CSS file in
+the file listing every style used in the UI components: `theme/components/_components.scss`.
+This will allow you to manage the import order of your stylesheets. Some more stylesheet
+entries can be found in `theme/modules/_modules.scss`, `theme/pages/_pages.scss` and
+`theme/main.scss`. Depending on the component you are styling, it might make more sense
+to use these stylesheets instead.
+
+<blockquote class="note">
+Please note here that we are using [Sass](https://sass-lang.com/) (hence the
+`.scss` extension). We believe that it is easier for developers new to the
+React Ecosystem to remain with this well-known CSS preprocessor.
+</blockquote>
+
+To import your own stylesheet in `theme/components/_components.scss`, you will need
+to override it from the base theme, like we did in [Extend the theme](extend-the-theme.html).
+
+```bash
+mkdir -p my-module/web/theme/components/
+cp node_modules/front-commerce/src/web/theme/components/_components.scss my-module/web/theme/components/_components.scss
+```
+
+Once it is done, you can add your stylesheet to the existing list.
+
+```scss
+// my-module/web/theme/components/_components.scss
+
+// ... import your IllustratedContent.scss at the end of this file
+import "./molecules/IllustratedContent/IllustratedContent";
+```
+
+```scss
+// my-module/web/theme/components/molecules/IllustratedContent/_IllustratedContent.scss
+.illustrated-content {
+  display: flex;
+  flex-direction: column;
+}
+.illustrated-content__media {
+  width: 30%;
+  max-width: 10em;
+}
+```
+
+If it didn't reload properly, it is because you need to restart the application
+every time you override a component in order to let the application know that the
+file location it should load has changed. But note that there is an upcoming
+improvement [#63](https://gitlab.com/front-commerce/front-commerce/issues/63) that
+should make things for you in the future.
+
+As a side note, we also use [BEM convention](http://getbem.com/naming/) for our
```

Comment:
We should not forget to add a link to our future « Advanced > Theme > BEM » page here when it will be live.

How to keep track of it?

Reply:
I'll add a TODO comment.